public/app/features/teams/state/reducers.ts

```diff
@@ -29,6 +29,13 @@ const teamsSlice = createSlice({
       const totalPages = Math.ceil(totalCount / perPage);
       return { ...state, ...rest, totalPages, perPage, hasFetched: true };
     },
+    userTeamsLoaded: (state, action: PayloadAction<Team[]>): TeamsState => {
+      const teams = [];
+      for (const key in action.payload) {
+        teams.push(action.payload[key]);
+      }
+      return { ...state, teams: teams, hasFetched: true };
```

Comment:
```suggestion
      return { ...state, teams: action.payload, hasFetched: true };
```
With the changes to `loadUserTeams`, this logic can be simplified.